src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp

```diff
@@ -152,6 +140,20 @@ void addONNXToZHighPasses(mlir::PassManager &pm) {
         onnx_mlir::zhigh::createZHighRecomposeToStickUnstickPass());
   }
 
+  // After all optimizations, if there are still light-weight ops (e.g. add,
+  // sub, ...) that are of `stick -> light-weight op -> unstick`, it's better to
+  // use CPU instead of NNPA to avoid stick/unstick. CPU is efficient to handle
+  // these ops, e.g vectorize the computation.
+  if (nnpaEnableZHighToOnnx)
+    pm.addNestedPass<func::FuncOp>(onnx_mlir::createZHighToONNXPass());
+
+  // Constant propagation at ZHighIR: constant stickify.
+  // Only support BE machines.
+  bool isBE = llvm::endianness::native == llvm::endianness::big;
+  if (isBE)
+    pm.addNestedPass<func::FuncOp>(
+        onnx_mlir::zhigh::createZHighConstPropagationPass());
```

Comment:
Move const propagation after recomposing `stick->nnpa_op->unstick` because once const propatation for stick is applied it's hard to recompose.

Reply:
NNPA is only for BE machines, so is that for having no issues on lit-tests?

Reply:
Yes, no issues on lit-tests. Lit tests for that constant propagation only run on BE machines.